avocado/utils/archive.py

```diff
@@ -39,6 +40,37 @@
         LZMA_CAPABLE = False
 
 
+#: The first two bytes that all gzip files start with
+GZIP_MAGIC = b'\037\213'
+
+
+def is_gzip_file(path):
+    """
+    Checks if file given by path has contents that suggests gzip file
+    """
+    with open(path, 'rb') as gzip_file:
+        return gzip_file.read(len(GZIP_MAGIC)) == GZIP_MAGIC
+
+
+def gzip_uncompress(path, output_path):
+    """
+    Uncompress a gzipped file at path, to either a file or dir at output_path
+    """
+    with gzip.GzipFile(filename=path, mode='rb') as input_file:
+        if os.path.isdir(output_path):
```

Comment:
What if the `output_path` be a file? This if is going to be false and the file content is going to be overwriten?

Reply:
Right, it tries to follow the semantics of many well know UNIX utilities, such as "cp" and "mv", while at the same time, respecting the suffix handling during decompression.  And yes, overwrites are the default when doing writes in all of those tools, functions, etc.

For this function, to have an overwrite, you should be giving the name of a previously existing file, so to me, it's like an intended action.

Reply:
Oh. It's really weird to me this kind of "distructive" behavior without a proper flag. You think it's reasonable go for it like this following others tools? If you're ok I'm ok :)

Reply:
Most (can't really say all) "write" operation in POSIX is destructive by default.  Just think of a simple `open('/boot/vmlinuz', 'w').write("GARBAGE")`, or the famous time that Linus "dialed" into his hard disk, instead of into his modem.  That's just he way things are! :smile: